### ISSUE TITLE
[Improvement] Improve liquidity scan for DEX trade, check both sides of the book

### ISSUE THREAD

**Issue by WietseWind** [XRPL-Labs/XUMM-Issue-Tracker#141]
Right now, we check for liquidity for the trade potentially happening. However, the other side of the order book may lack liquidity. As a result eg. an asset can be purchased, but if someone wants to undo the trade, a very bad rate will be available for the return path. If this is the case, we should not allow the trade to go through in the first place.

So:
- Instead of checking the (to use) side of the order book, check both sides for the same amount
- Check if there are enough offers by different accounts
- Check if the reverse order would not have a price offset > X %

Sample bad order book:
![image](https://user-images.githubusercontent.com/4756161/79986852-f6ea1a00-84ac-11ea-81ac-abdf17f005f6.png)

In this case buying is easy, but returning the funds will result in a big loss pretty damn fast. We should protect users.

# What to change
### For the EXCHANGE button for an IOU (home screen):
- Use the [reader](https://www.npmjs.com/package/xrpl-orderbook-reader) to get price based on the entered amount, if changed: get new price. Periodically update if users keep the screen open and liquidity changes.
- Implement (Params, options, see below) liquidity boundries, and handle Errors as returned by the [reader](https://www.npmjs.com/package/xrpl-orderbook-reader)
- Keep the message to use XRPToolkit for advanced users #208 

### For the SEND / Request flow where liquidity checks are in place (in case conversion/partial payment will happen)
See `For the EXCHANGE button for an IOU (home screen)` (above), and respect existing conditions for applying (allowing / denying) exchange (based on amount within spendable balance, etc).

### Params (options with boundries/check limits)
To check the params (options) to use for the liquiduty checker, fetch the issuer+IOU specific params from the XUMM backend:
`/api/v1/app/liquidity-boundries/<issuer>/<IOU>`
eg.
/api/v1/app/liquidity-boundries/rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq/EUR

Response:
```
{
  "issuer": "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq",
  "iou": "EUR",
  "options": {
    "rates": "to",
    "minOfferCount": 10,
    "maxSpreadPercentage": 4,
    "maxSlippagePercentage": 2,
    "maxSlippagePercentageReverse": 5,
    "maxBookLines": 500
  }
}
```

The `options` property can be used (eg with spread operator) directly into the `Params`»`options` for the liquidity checker.

## Error handling (to client)
I think we should have just one generic liquidity error, with an info link/icon/button to show the underlying reason(s, multiple possible):
https://github.com/XRPL-Labs/XRPL-Orderbook-Reader/blob/master/src/types/Reader.ts#L9
-  `REQUESTED_LIQUIDITY_NOT_AVAILABLE` The amount can't be exchanged: there are not enough offers available on the Decentralized Exchange.
-  `REVERSE_LIQUIDITY_NOT_AVAILABLE` The amount can't be exchanged: based on the lack of offers on the Decentralized Exchange, you would not be able to exchange it back.
-  `MAX_SPREAD_EXCEEDED` The price difference between selling and buying offers is too high.
-  `MAX_SLIPPAGE_EXCEEDED` The exchange rate will be bad as there are not enough (good) offers on the Decentralized Exchange.
-  `MAX_REVERSE_SLIPPAGE_EXCEEDED` The exchange rate will be bad if you would try to exchange back, as there are not enough (good) offers on the Decentralized Exchange.
-  `INSUFFICIENT_AMOUNT_OF_OFFERS` There are not enough offers available on the Decentralized Exchange for this pair.

**Comment by WietseWind** [comment]
Links to #98

**Comment by WietseWind** [comment]
https://github.com/XRPL-Labs/XRPL-Orderbook-Reader 🎉

**Comment by WietseWind** [comment]
Fixes #98
